Src/BootCamp.Chapter/ContactsCenter.cs

```diff
@@ -1,66 +1,108 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
-using System.Text;
 
 namespace BootCamp.Chapter
 {
     public class ContactsCenter
     {
-         private readonly string _peopleFile;
+        private readonly string _peopleFile;
 
         public ContactsCenter(string peopleFile)
         {
             if (!File.Exists(peopleFile))
             {
-                throw new FileNotFoundException("File is made up"); 
+                throw new FileNotFoundException("File is made up");
             }
-                      
-            _peopleFile = File.ReadAllText(peopleFile); 
-           
+
+            _peopleFile = File.ReadAllText(peopleFile);
+
             if (String.IsNullOrWhiteSpace(_peopleFile))
             {
-                throw new Exception(); 
+                throw new InvalidOperationException("File cannot be empty");
             }
-         
         }
 
-       
         /// <summary>
         /// Gets people by filter predicate.
         /// </summary>
         /// <returns></returns>
         public List<Person> Filter(Predicate<Person> predicate)
         {
             var people = new List<Person>();
-            var splittedList = _peopleFile.Split($"{Environment.NewLine}");
-            for (int i = 1; i < splittedList.Length -1 ; i++)
+            var splittedList = _peopleFile.Split(Environment.NewLine);
+            // skipped the first row because its a header.
+            for (int i = 1; i < splittedList.Length - 1; i++)
             {
-                var person = ConvertStringToPerson(splittedList[i]);
+                var person = TryParse(splittedList[i]);
```

Comment:
Still this is not the tryparse pattern you did in previous lessons.

Reply:
hmm, then I have to look what I have to change. The earlier tryparse were simpler for example look if a string could be parsed to a int. This is parsing a string to a object.

Reply:
Right now I have this : 
`` 
public Person TryParse(string personString)
        {
            
            CultureInfo culture = new CultureInfo("en-US");
            var splittedPerson = personString.Split(',');

            var isValid = DateTime.TryParse(splittedPerson[2], culture, DateTimeStyles.None, out DateTime date);
            if (!isValid)
            {
                throw new ArgumentException("date is not valid");
            }

            isValid = Enum.TryParse(splittedPerson[3], true, out Gender gender);

            if (!isValid)
            {
                throw new ArgumentException("Gender can only be Male or Female");
            }

            var person = new Person();
            person.Name = splittedPerson[0];
            person.SureName = splittedPerson[1];
            person.BirthDay = date;
            person.Gender = gender;
            person.Country = splittedPerson[4];
            person.Email = splittedPerson[5];
            person.StreetAdress = splittedPerson[6];

            return person;
        }
 ```

Reply:
you should out a person like in a tryparse pattern. This should be a part of a person class.